test/edu/stanford/dlss/was/TestWasapiConnection.java

```diff
@@ -33,6 +37,24 @@ public void jsonQueryCallsExecute() throws IOException {
   }
 
   @Test
+  public void pagedJsonQuery_loopsThroughAndReturnsResponseList() throws IOException {
+    WasapiClient mockClient = mock(WasapiClient.class);
+    WasapiConnection spyConnection = spy(new WasapiConnection(mockClient));
+
+    WasapiResponse[] responses = {
+      mockResponseNotLast(),
+      mockResponseLast()
+    };
+    doReturn(responses[0]).when(spyConnection).jsonQuery(ORIG_QUERY_URL);
+    doReturn(responses[1]).when(spyConnection).jsonQuery(NEXT_URL);
+
+    List<WasapiResponse> respList = spyConnection.pagedJsonQuery(ORIG_QUERY_URL);
+    assertEquals("response list should have 2 entries", 2, respList.size());
+    assertEquals("response list's first entry should be first response", responses[0], respList.get(0));
+    assertEquals("response list's second entry should be second response", responses[1], respList.get(1));
+  }
+
```

Comment:
added tests for a null `jsonQuery` result (and refactored slightly to have `jsonQuery` handle that situation more gracefully itself, which didn't break any of the existing tests, but also added a test for that specific scenario, see new first commit on this branch).  also added a test for a single page result for `pagedJsonQuery`.